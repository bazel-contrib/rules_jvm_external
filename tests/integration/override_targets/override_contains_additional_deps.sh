# --- begin runfiles.bash initialization v2 ---
# Copy-pasted from the Bazel Bash runfiles library v2.
set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
  source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
  source "$0.runfiles/$f" 2>/dev/null || \
  source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
  source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
  { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
# --- end runfiles.bash initialization v2 ---

set -euox pipefail

deps_file=$(rlocation rules_jvm_external/tests/integration/override_targets/trace_otel_deps)

# we should contain the original target
if ! grep -q "@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk" $deps_file; then
  echo "Unable to find SDK target"
  exit 1
fi

# should contain the "raw" dep
if ! grep -q "@override_target_in_deps//:original_io_opentelemetry_opentelemetry_api" $deps_file; then
  echo "Unable to find raw API target"
  exit 1
fi

# the "context" dependency is depended upon by `io.opentelemetry:opentelemetry-api` and
# nothing else in the SDK. If we have built the raw dependency properly, this should
# also be present in the dependencies
if ! grep -q "@override_target_in_deps//:io_opentelemetry_opentelemetry_context" $deps_file; then
  echo "Unable to find transitive dep of raw target"
  exit 1
fi

# Finally, we expect jedis (which is not an OTel dep) to have also been added
if ! grep -q "@override_target_in_deps//:redis_clients_jedis" $deps_file; then
  echo "Unable to find additional target added to a transitive dep of the SDK"
  exit 1
fi
