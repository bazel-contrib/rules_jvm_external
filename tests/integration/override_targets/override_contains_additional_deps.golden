//tests/integration/override_targets:additional_deps
@bazel_tools//src/conditions:host_windows
@bazel_tools//src/conditions:host_windows_arm64_constraint
@bazel_tools//src/conditions:host_windows_x64_constraint
@com_google_code_gson_gson_2_10_1//file:file
@com_google_code_gson_gson_2_10_1//file:v1/com/google/code/gson/gson/2.10.1/gson-2.10.1.jar
@io_opentelemetry_opentelemetry_api_1_28_0//file:file
@io_opentelemetry_opentelemetry_api_1_28_0//file:v1/io/opentelemetry/opentelemetry-api/1.28.0/opentelemetry-api-1.28.0.jar
@io_opentelemetry_opentelemetry_api_events_1_28_0_alpha//file:file
@io_opentelemetry_opentelemetry_api_events_1_28_0_alpha//file:v1/io/opentelemetry/opentelemetry-api-events/1.28.0-alpha/opentelemetry-api-events-1.28.0-alpha.jar
@io_opentelemetry_opentelemetry_context_1_28_0//file:file
@io_opentelemetry_opentelemetry_context_1_28_0//file:v1/io/opentelemetry/opentelemetry-context/1.28.0/opentelemetry-context-1.28.0.jar
@io_opentelemetry_opentelemetry_extension_incubator_1_28_0_alpha//file:file
@io_opentelemetry_opentelemetry_extension_incubator_1_28_0_alpha//file:v1/io/opentelemetry/opentelemetry-extension-incubator/1.28.0-alpha/opentelemetry-extension-incubator-1.28.0-alpha.jar
@io_opentelemetry_opentelemetry_sdk_1_28_0//file:file
@io_opentelemetry_opentelemetry_sdk_1_28_0//file:v1/io/opentelemetry/opentelemetry-sdk/1.28.0/opentelemetry-sdk-1.28.0.jar
@io_opentelemetry_opentelemetry_sdk_common_1_28_0//file:file
@io_opentelemetry_opentelemetry_sdk_common_1_28_0//file:v1/io/opentelemetry/opentelemetry-sdk-common/1.28.0/opentelemetry-sdk-common-1.28.0.jar
@io_opentelemetry_opentelemetry_sdk_logs_1_28_0//file:file
@io_opentelemetry_opentelemetry_sdk_logs_1_28_0//file:v1/io/opentelemetry/opentelemetry-sdk-logs/1.28.0/opentelemetry-sdk-logs-1.28.0.jar
@io_opentelemetry_opentelemetry_sdk_metrics_1_28_0//file:file
@io_opentelemetry_opentelemetry_sdk_metrics_1_28_0//file:v1/io/opentelemetry/opentelemetry-sdk-metrics/1.28.0/opentelemetry-sdk-metrics-1.28.0.jar
@io_opentelemetry_opentelemetry_sdk_trace_1_28_0//file:file
@io_opentelemetry_opentelemetry_sdk_trace_1_28_0//file:v1/io/opentelemetry/opentelemetry-sdk-trace/1.28.0/opentelemetry-sdk-trace-1.28.0.jar
@io_opentelemetry_opentelemetry_semconv_1_28_0_alpha//file:file
@io_opentelemetry_opentelemetry_semconv_1_28_0_alpha//file:v1/io/opentelemetry/opentelemetry-semconv/1.28.0-alpha/opentelemetry-semconv-1.28.0-alpha.jar
@org_apache_commons_commons_pool2_2_11_1//file:file
@org_apache_commons_commons_pool2_2_11_1//file:v1/org/apache/commons/commons-pool2/2.11.1/commons-pool2-2.11.1.jar
@org_json_json_20231013//file:file
@org_json_json_20231013//file:v1/org/json/json/20231013/json-20231013.jar
@org_slf4j_slf4j_api_1_7_36//file:file
@org_slf4j_slf4j_api_1_7_36//file:v1/org/slf4j/slf4j-api/1.7.36/slf4j-api-1.7.36.jar
@override_target_in_deps//:com/google/code/gson/gson/2.10.1/gson-2.10.1.jar
@override_target_in_deps//:com_google_code_gson_gson
@override_target_in_deps//:com_google_code_gson_gson_2_10_1_extension
@override_target_in_deps//:io/opentelemetry/opentelemetry-api-events/1.28.0-alpha/opentelemetry-api-events-1.28.0-alpha.jar
@override_target_in_deps//:io/opentelemetry/opentelemetry-api/1.28.0/opentelemetry-api-1.28.0.jar
@override_target_in_deps//:io/opentelemetry/opentelemetry-context/1.28.0/opentelemetry-context-1.28.0.jar
@override_target_in_deps//:io/opentelemetry/opentelemetry-extension-incubator/1.28.0-alpha/opentelemetry-extension-incubator-1.28.0-alpha.jar
@override_target_in_deps//:io/opentelemetry/opentelemetry-sdk-common/1.28.0/opentelemetry-sdk-common-1.28.0.jar
@override_target_in_deps//:io/opentelemetry/opentelemetry-sdk-logs/1.28.0/opentelemetry-sdk-logs-1.28.0.jar
@override_target_in_deps//:io/opentelemetry/opentelemetry-sdk-metrics/1.28.0/opentelemetry-sdk-metrics-1.28.0.jar
@override_target_in_deps//:io/opentelemetry/opentelemetry-sdk-trace/1.28.0/opentelemetry-sdk-trace-1.28.0.jar
@override_target_in_deps//:io/opentelemetry/opentelemetry-sdk/1.28.0/opentelemetry-sdk-1.28.0.jar
@override_target_in_deps//:io/opentelemetry/opentelemetry-semconv/1.28.0-alpha/opentelemetry-semconv-1.28.0-alpha.jar
@override_target_in_deps//:io_opentelemetry_opentelemetry_api_1_28_0_extension
@override_target_in_deps//:io_opentelemetry_opentelemetry_api_events
@override_target_in_deps//:io_opentelemetry_opentelemetry_api_events_1_28_0_alpha_extension
@override_target_in_deps//:io_opentelemetry_opentelemetry_context
@override_target_in_deps//:io_opentelemetry_opentelemetry_context_1_28_0_extension
@override_target_in_deps//:io_opentelemetry_opentelemetry_extension_incubator
@override_target_in_deps//:io_opentelemetry_opentelemetry_extension_incubator_1_28_0_alpha_extension
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk_1_28_0_extension
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk_common
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk_common_1_28_0_extension
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk_logs
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk_logs_1_28_0_extension
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk_metrics
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk_metrics_1_28_0_extension
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk_trace
@override_target_in_deps//:io_opentelemetry_opentelemetry_sdk_trace_1_28_0_extension
@override_target_in_deps//:io_opentelemetry_opentelemetry_semconv
@override_target_in_deps//:io_opentelemetry_opentelemetry_semconv_1_28_0_alpha_extension
@override_target_in_deps//:org/apache/commons/commons-pool2/2.11.1/commons-pool2-2.11.1.jar
@override_target_in_deps//:org/json/json/20231013/json-20231013.jar
@override_target_in_deps//:org/slf4j/slf4j-api/1.7.36/slf4j-api-1.7.36.jar
@override_target_in_deps//:org_apache_commons_commons_pool2
@override_target_in_deps//:org_apache_commons_commons_pool2_2_11_1_extension
@override_target_in_deps//:org_json_json
@override_target_in_deps//:org_json_json_20231013_extension
@override_target_in_deps//:org_slf4j_slf4j_api
@override_target_in_deps//:org_slf4j_slf4j_api_1_7_36_extension
@override_target_in_deps//:original_io_opentelemetry_opentelemetry_api
@override_target_in_deps//:redis/clients/jedis/5.0.2/jedis-5.0.2.jar
@override_target_in_deps//:redis_clients_jedis
@override_target_in_deps//:redis_clients_jedis_5_0_2_extension
@redis_clients_jedis_5_0_2//file:file
@redis_clients_jedis_5_0_2//file:v1/redis/clients/jedis/5.0.2/jedis-5.0.2.jar
